eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -499,6 +496,40 @@ public function buildLocationWithContent(
         return $this->mapLocation($spiLocation, $contentInfo, $content);
     }
 
+    /**
+     * Builds API Location object for tree root.
+     *
+     * @param \eZ\Publish\SPI\Persistence\Content\Location $spiLocation
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Location
+     */
+    private function buildRootLocation(SPILocation $spiLocation): APILocation
+    {
+        //  first known commit of eZ Publish 3.x
+        $legacyDateTime = $this->getDateTime(1030968000);
+
+        // NOTE: this is hardcoded workaround for missing ContentInfo on root location
+        return $this->mapLocation(
+            $spiLocation,
+            new ContentInfo([
+                'id' => 0,
+                'name' => 'Top Level Nodes',
+                'sectionId' => 1,
+                'mainLocationId' => 1,
+                'contentTypeId' => 1,
+                'currentVersionNo' => 1,
+                'published' => 1,
+                'ownerId' => 14, // admin user
```

Comment:
what if you delete admin user and use your own?

Reply:
I asked exactly the same question when I saw this code snippet for the first time 😞 This definitely shouldn't be hardcoded, but since this PR is about aligning behavior of the `\eZ\Publish\Core\Repository\Helper\DomainMapper::buildLocation` and `\eZ\Publish\Core\Repository\Helper\DomainMapper::buildLocationWithContent` so this ugly code snippet need to stay with us litte bit longer. I created follow up ticket for this:  https://jira.ez.no/browse/EZP-30309

Reply:
The root of the issue is that there is no content for this root location in database, until there is there is no point in trying to fix these things _(language, owner, ..)_. It will just be ugly workarounds on a workaround ;)

Once there is content in the database for this, we'll need to figure out how to lock it from being wiped on mistake and so on....